code/modules/awaymissions/mission_code/UO71-terrorspiders.dm

```diff
@@ -0,0 +1,319 @@
+	<p>Purple<br>Only seen guarding the nest of the Queen to the south. Appear to be territorial, and very dangerous.</p>\
+	<p>Prince<br>Held in containment. Appears to be a sort of super-warrior. Fast, strong, and extremely resistant to damage.</p>\
+	<p>Mother<br>Passive mother of many young. Also held in containment. Appears benign if undisturbed.</p>\
+	<p>Queen<br>Unable to contain. Present south of Cargo before contact was lost. Presumed ruler of the local hive.</p>\
+	"
+
+/obj/item/weapon/paper/terrorspiders6
+	name = "paper= 'A Study in Venom'"
+	info = "<b>Initial Report - Black Widow Venom</b><br>\
+	<br>\
+	Even a single bite kills within a few minutes. Immediate charcoal is necessary for the victim to survive.<br>\
+	The fascinating thing is that multiple bites actually cause rapid symptom onset. The victim quickly loses the ability to walk.<br>\
+	One test monkey went from perfect health to death in under 30 seconds - from 3 bites.<br>\
+	I am so very glad I'm not responsible for keeping these things contained."
+
+/obj/item/weapon/paper/terrorspiders7
+	name = "paper= 'Security Orders'"
+	info = "<b>Directives for Officer James</b><br>\
+	<br>\
+	1. Find a way to lift the bridge lockdown.<br>\
+	2. Kill any spiders you encounter.<br>\
+	3. Once on the bridge, open a path to the Queen in the south, and find a way to kill her, ending the infestation. <br>\
+	Good luck, Officer.<br>\
+	"
+
+/obj/item/weapon/paper/terrorspiders8
+	name = "paper= 'Last Words'"
+	info = "<b>The Last Words of Nurse Barnes</b><br>\
+	<br>\
+	It is too much. First Phil, then those damn spiders bite Mary too. Heaven help us. <br>\
+	I'll be evacuating as soon as I've grabbed my supplies. I had to pay for those myself. Stupid budget cuts. <br>\
+	The new world they've found for us isn't that hospitable, but at least there are no deadly spid---<br>\
+	(the writing trails off, as if the writer was interrupted)<br>\
+	"
+
+
+/obj/item/weapon/gun/energy/awaymission_aeg
+	name = "Wireless Energy Gun"
+	desc = "An energy gun that recharges wirelessly during away missions. Does not work on the main station."
+	icon_state = "nucgun"
+	projectile_type = "/obj/item/projectile/beam"
+	fire_sound = 'sound/weapons/Laser.ogg'
+	origin_tech = "combat=5;materials=5;powerstorage=5;engineering=3;"
+	var/inawaymission = 1
+	can_flashlight = 0
+	can_charge = 0
+	charge_cost = 1000
+
+/obj/item/weapon/gun/energy/awaymission_aeg/New()
+	..()
+	processing_objects.Add(src)
+
+/obj/item/weapon/gun/energy/awaymission_aeg/Destroy()
+	processing_objects.Remove(src)
+	return ..()
+
+/obj/item/weapon/gun/energy/awaymission_aeg/process()
+	var/turf/my_loc = get_turf(src)
+	if(my_loc.z == 8)
+		if (inawaymission)
+			charge_tick++
+			if(charge_tick < 4) return 0
+			charge_tick = 0
+			if(!power_supply) return 0
+			if((power_supply.charge / power_supply.maxcharge) != 1)
+				power_supply.give(1000)
+				update_icon()
+			return 1
```

Comment:
either remove this return or remove the else below, you shouldn't use else statements *and* early returns